ddp-workspace/projects/ddp-dsm-ui/src/app/STORE/effects/participants.effects.ts

```diff
@@ -0,0 +1,35 @@
+import {Injectable} from '@angular/core';
+import {Actions, createEffect, ofType} from '@ngrx/effects';
+import {map, exhaustMap, catchError} from 'rxjs/operators';
+import * as ParticipantActions from '../actions/participants.actions';
+import {DSMService} from '../../services/dsm.service';
+import {of} from 'rxjs';
+
+@Injectable()
+export class ParticipantsEffects {
+  constructor(
+    private actions$: Actions,
+    private dsmService: DSMService,
+  ) {
+  }
+
+  getParticipants$ = createEffect(() => this.actions$.pipe(
+      ofType(ParticipantActions.getParticipantsRequest),
+      exhaustMap(({viewFilter, realm, parent, filterQuery, from, to, sort}) =>
+        this.dsmService.applyFilter(viewFilter, realm, parent, filterQuery, from, to, sort)
+          .pipe(
```

Comment:
This is the file where we control HTTP requests (their frequency, flow, state and etc.).